tests/kernel/usb_nic.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: usb_nic
+# Summary: Simple smoke test for testing USB NIC connected to system
+# Maintainer: LSG QE Kernel <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_serial_terminal;
+
+    my $interface = script_output 'basename $(readlink /sys/class/net/* | grep usb )';
+
+    assert_script_run "echo \"BOOTPROTO='dhcp'\" > /etc/sysconfig/network/ifcfg-$interface";
+    assert_script_run "echo \"STARTMODE='auto'\" >> /etc/sysconfig/network/ifcfg-$interface";
+
+    assert_script_run("ifup $interface -o debug", 60);
+
+    my $route = script_output "ip route show default";
+    my @ping_peer = $route =~ /default via (\S+)/;
+    assert_script_run "ping -I $interface -c 4 @ping_peer";
```

Comment:
Depending on network structure, this may not work. The default gateway returned by `ip route show default` may be reachable only through the onboard NIC.

Reply:
That's why you can also specify the ping peer by a variable - we just try to find something to ping in case you don't have a specific target.

Reply:
You could also read `ip neigh show dev $interface` and then try to ping each peer until at least one responds.

Reply:
We are developing test with just for one machine in one physical network setup. Both network interfaces are connected to same network and share same gateway and this will not change.

I believe, we don't need to cover other non existing scenario.